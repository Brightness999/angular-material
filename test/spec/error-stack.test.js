/**
 * @fileOverview Test logging a JS error with error stack.
 */
const Logality = require('../..');
const { sink, stubLogality } = require('../lib/tester.lib');

describe('Error Stack Testing', () => {
  stubLogality();

  test('Will properly figure out invoking function module on error', (done) => {
    const logality = new Logality({
      appName: 'testLogality',
      wstream: sink((chunk) => {
        expect(chunk).toHaveProperty('severity', 3);
        expect(chunk).toHaveProperty('message', 'An error happened, maybe?');
        expect(chunk).toHaveProperty('level', 'error');
        expect(chunk).toHaveProperty('event');
        expect(chunk).toHaveProperty('dt', '2018-05-18T16:25:57.815Z');
        expect(chunk).toHaveProperty('context');

        expect(chunk.event).toHaveProperty('error');
        expect(chunk.event.error).toHaveProperty('backtrace');
        expect(chunk.event.error).toHaveProperty('message', 'An error did not occur');
        expect(chunk.event.error).toHaveProperty('name', 'Error');

        expect(chunk.event.error.backtrace.length).toBeGreaterThan(4);

        chunk.event.error.backtrace.forEach((traceItem) => {
          expect(traceItem).toHaveProperty('file');
          expect(traceItem).toHaveProperty('function');
          expect(traceItem).toHaveProperty('line');
        });
        done();
      }),
    });

    const log = logality.get();

    try {
      throw new Error('An error did not occur');
    } catch (ex) {
      log('error', 'An error happened, maybe?', {
        error: ex,
      });
    }
  });
});
