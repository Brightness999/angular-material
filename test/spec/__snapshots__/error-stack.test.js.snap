// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Stack Testing Will properly figure out invoking function module on error 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
      "file": "/test/spec/error-stack.test.js",
    },
    "source": Object {
      "file_name": "/test/spec/error-stack.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {
    "error": Object {
      "backtrace": Array [
        Object {
          "file": "/Users/thanpolas/Projects/work/alacrity/logality/test/spec/error-stack.test.js",
          "function": "Object.done",
          "line": "22:13",
        },
        Object {
          "file": "/Users/thanpolas/Projects/work/alacrity/logality/node_modules/jest-jasmine2/build/queue_runner.js",
          "function": "resolve",
          "line": "52:12",
        },
        Object {
          "file": null,
          "function": "new Promise",
          "line": "null:null",
        },
        Object {
          "file": "/Users/thanpolas/Projects/work/alacrity/logality/node_modules/jest-jasmine2/build/queue_runner.js",
          "function": "mapper",
          "line": "39:19",
        },
        Object {
          "file": "/Users/thanpolas/Projects/work/alacrity/logality/node_modules/jest-jasmine2/build/queue_runner.js",
          "function": "promise.then",
          "line": "73:82",
        },
      ],
      "message": "An error did not occur",
      "name": "Error",
    },
  },
  "level": "error",
  "message": "An error happened, maybe?",
  "severity": 3,
}
`;
